### DISCUSSION TITLE
Reference Variable Java Development

### DISCUSSION THREAD

**Discussion by srinivasaraogurram** [community/community#120124] [Programming Help]
### Select Topic Area

Product Feedback

### Body

Hi VS Code Team, I am looking  variable reference
https://code.visualstudio.com/docs/editor/variables-reference
I am using Code Runner for Running Java Code. 
When Run the JavaCode, it is excepting PackageName followed by Class Name to Run the Java Code
"code-runner.executorMap": {
    "java": "cd $dir && javac $fileName && java -classpath . $dir$fileNameWithoutExt",
}
The above option is giving following Error
```
[Running] cd "/Users/sri/Downloads/JavaWithPackage/com/coderunner/javaapp/" && javac HelloWorld.java && java -classpath . "/Users/sri/Downloads/JavaWithPackage/com/coderunner/javaapp/"HelloWorld
Error: Could not find or load main class .Users.sri.Downloads.JavaWithPackage.com.coderunner.javaapp.HelloWorld
Caused by: java.lang.ClassNotFoundException: /Users/sri/Downloads/JavaWithPackage/com/coderunner/javaapp/HelloWorld
```
as show in the screenshot
![image](https://github.com/community/community/assets/2825912/1b977c28-0050-482b-9e28-8c25be874f22)
In order to change, I need to pass as com.coderunner.javaaapp.HelloWorld
If Team can come up with a reference variables where I can get $directorypathwithfilenamewithdotseperated 
this will help to extend code runner to JavaProjects with Packages

Error
![image](https://github.com/community/community/assets/2825912/20ccc390-5cba-490f-8507-0430fb78e50e)

**Comment by ebndev** [comment]
Hi @srinivasaraogurram,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help, you might want to try posting this feedback somewhere that focuses on VS Code support.

The GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!

**Comment by yangwei980131** [comment]
Configure the JDK environment
